### PR TITLE
fix: correct default dispatch throttle enabled

### PR DESCRIPTION
**Pull request by adriantam**
## Description

Initialization of server dispatch throttle enabled should use default dispatch throttle enabled

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		requestDurationByDispatchCountHistogramBuckets: []uint{50, 200},
 		serviceName: openfgav1.OpenFGAService_ServiceDesc.ServiceName,
 
-		dispatchThrottlingCheckResolverEnabled:   serverconfig.DefaultCheckQueryCacheEnable,
+		dispatchThrottlingCheckResolverEnabled:   serverconfig.DefaultDispatchThrottlingEnabled,
```
Comment (jpadilla) [inline pkg/server/server.go:366]: uff good catch.

**Review by jpadilla — APPROVED** [review]

File: pkg/server/server.go (hunk 1 above)
Comment (miparnisari) [reply pkg/server/server.go:366]: how did we miss this? should we update the test `dispatch_throttling_check_resolver_enabled`?

Comment (adriantam) [reply pkg/server/server.go:366]: Default values are constant and they have the same value (false).  I don't know if we can override constants in the UT.

Comment (miparnisari) [reply pkg/server/server.go:366]: ah, i see. yeah not sure how to test for that..

**Review by miparnisari — APPROVED** [review]

**Pull request merged**